plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,421 @@
+	"extendwrite",
+	"sillyrenames",
+	"shortreads",
+	"shortwrites",
+	"delay",
+	"pnfsreads",
+	"pnfswrites",
+}
+
+var bytesFields = []string{
+	"normalreadbytes",
+	"normalwritebytes",
+	"directreadbytes",
+	"directwritebytes",
+	"serverreadbytes",
+	"serverwritebytes",
+	"readpages",
+	"writepages",
+}
+
+var xprtudpFields = []string{
+	"bind_count",
+	"rpcsends",
+	"rpcreceives",
+	"badxids",
+	"inflightsends",
+	"backlogutil",
+}
+
+var xprttcpFields = []string{
+	"bind_count",
+	"connect_count",
+	"connect_time",
+	"idle_time",
+	"rpcsends",
+	"rpcreceives",
+	"badxids",
+	"inflightsends",
+	"backlogutil",
+}
+
+var nfs3Fields = []string{
+	"NULL",
+	"GETATTR",
+	"SETATTR",
+	"LOOKUP",
+	"ACCESS",
+	"READLINK",
+	"READ",
+	"WRITE",
+	"CREATE",
+	"MKDIR",
+	"SYMLINK",
+	"MKNOD",
+	"REMOVE",
+	"RMDIR",
+	"RENAME",
+	"LINK",
+	"READDIR",
+	"READDIRPLUS",
+	"FSSTAT",
+	"FSINFO",
+	"PATHCONF",
+	"COMMIT",
+}
+
+var nfs4Fields = []string{
+	"NULL",
+	"READ",
+	"WRITE",
+	"COMMIT",
+	"OPEN",
+	"OPEN_CONFIRM",
+	"OPEN_NOATTR",
+	"OPEN_DOWNGRADE",
+	"CLOSE",
+	"SETATTR",
+	"FSINFO",
+	"RENEW",
+	"SETCLIENTID",
+	"SETCLIENTID_CONFIRM",
+	"LOCK",
+	"LOCKT",
+	"LOCKU",
+	"ACCESS",
+	"GETATTR",
+	"LOOKUP",
+	"LOOKUP_ROOT",
+	"REMOVE",
+	"RENAME",
+	"LINK",
+	"SYMLINK",
+	"CREATE",
+	"PATHCONF",
+	"STATFS",
+	"READLINK",
+	"READDIR",
+	"SERVER_CAPS",
+	"DELEGRETURN",
+	"GETACL",
+	"SETACL",
+	"FS_LOCATIONS",
+	"RELEASE_LOCKOWNER",
+	"SECINFO",
+	"FSID_PRESENT",
+	"EXCHANGE_ID",
+	"CREATE_SESSION",
+	"DESTROY_SESSION",
+	"SEQUENCE",
+	"GET_LEASE_TIME",
+	"RECLAIM_COMPLETE",
+	"LAYOUTGET",
+	"GETDEVICEINFO",
+	"LAYOUTCOMMIT",
+	"LAYOUTRETURN",
+	"SECINFO_NO_NAME",
+	"TEST_STATEID",
+	"FREE_STATEID",
+	"GETDEVICELIST",
+	"BIND_CONN_TO_SESSION",
+	"DESTROY_CLIENTID",
+	"SEEK",
+	"ALLOCATE",
+	"DEALLOCATE",
+	"LAYOUTSTATS",
+	"CLONE",
+}
+
+var nfsopFields = []string{
+	"ops",
+	"trans",
+	"timeouts",
+	"bytes_sent",
+	"bytes_recv",
+	"queue_time",
+	"response_time",
+	"total_time",
+	"errors",
+}
+
+func convert(line []string) []int64 {
+	var nline []int64
+	for _, l := range line[1:] {
+		f, _ := strconv.ParseInt(l, 10, 64)
+		nline = append(nline, f)
+	}
+	return nline
+}
+
+func in(list []string, val string) bool {
+	for _, v := range list {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *NFSClient) parseStat(mountpoint string, export string, version string, line []string, fullstat bool, nfs3Ops map[string]bool, nfs4Ops map[string]bool, acc telegraf.Accumulator) error {
+	tags := map[string]string{"mountpoint": mountpoint, "serverexport": export}
+	nline := convert(line)
+	first := strings.Replace(line[0], ":", "", 1)
+
+	var fields = make(map[string]interface{})
+
+	if fullstat && first == "events" && len(nline) >= len(eventsFields) {
+		for i, t := range eventsFields {
+			fields[t] = nline[i]
+		}
+		acc.AddFields("nfs_events", fields, tags)
+	} else if fullstat && first == "bytes" && len(nline) >= len(bytesFields) {
```

Comment:
With the reorganization above you might want to `switch` on `first`...
I'm also interested in the length-check. Is this an error case or a "normal-operation" to have a line with too few fields?

Reply:
Actually, this is already done, just not in this set of changes (I've since pushed them to my branch, and will see about adding to this PR).  The "non-fullstat" metrics are always returned, and a simple "if fullstat {}" block encompasses the rest of the parsing.

>  I'm also interested in the length-check. Is this an error case or a "normal-operation" to have a line with too few fields?

Yes.  I hit at least one case where there there were missing entries at the end of certain fields.  This is true in the "per-op statistics" at a minimum, where older kernels (something around 3.10.x or older) have 8 data fields:
```
        per-op statistics
                NULL: 1 1 0 44 24 0 0 0
```
Newer kernels have 9:
```
        per-op statistics
                NULL: 1 1 0 12 8 0 0 0 0
```
 
And yes, there are use cases for those older kernels (unfortunately).

Reply:
Hi, I merged your change into the brach from the PR. As mentioned before I gave you access to commit straight on my fork (git@github.com:pmoranga/telegraf.git), feel free to use.

Reply:
The per-op metric counts added a field in recent kernels:  9 columns of data instead of 8, so we need to account for this, at least in that case.